app/helpers/view_components/site_footer.rb

```diff
@@ -18,5 +15,26 @@ def to_s
         end
       end
     end
+
+    private
+    def cache_key
+      parts = digests
+      parts << ::Track.active.count
+      parts << user_part
+
+      parts.join(':')
+    end
+
+    def user_part
+      return 0 unless user_signed_in?
+
+      current_user.captcha_required? ? 2 : 1
+    end
```

Comment:
This ensures that the cache includes whether or not the user is required for a captcha to be shown